### PR TITLE
spec file BuildConstraints examples

### PR DESCRIPTION
**Pull request by adrianschroeter**

### CODE REVIEW

**Review by hennevogel — CHANGES_REQUESTED** [review]

**Comment by tserong** [comment]
I've been experimenting with `#!BuildConstraint` today, and it looks like if a `_constraints` file is present, the values in that file will override the values from `#!BuildConstraint` in spec files. Maybe also add a note about values from `_constraints` taking precedence over what's in the spec files?

**Comment by tserong** [comment]
Oh, also, `osc checkconstraints` seems to completely ignore `#!BuildConstraint`. Should that be documented too, or shall I go open a bug somewhere?

Thanks!

**Comment by darix** [comment]
This PR should probably mention that you shouldnt mix _constriants and `#!BuildConstraints`

**Comment by adrianschroeter** [comment]
> I've been experimenting with `#!BuildConstraint` today, and it looks like if a `_constraints` file is present, the values in that file will override the values from `#!BuildConstraint` in spec files. Maybe also worth adding a note about values from `_constraints` taking precedence over what's in the spec files?

the opposite is true meanwhile, it is part of the docu now

**Comment by adrianschroeter** [comment]
> This PR should probably mention that you shouldnt mix _constriants and `#!BuildConstraint`

it is fine to combine that. It is not different as with mixing from project build config.

File: xml/obs_build_job_constraints.xml [hunk 1]
```diff
@@ -19,6 +19,10 @@
  <para> The build constraints for a package are part of the package sources, as
   a _constraints XML source file (validated on submission). The _constraints
   source file can contain the values listed below. </para>
+ <para>In addition to that some build receipe parsers support to parse constraints
+  also from the build descriptions, eg. in spec files. These can also be used
```
[the remark this answers is not in the document]
Comment (adrianschroeter) [reply xml/obs_build_job_constraints.xml:23]: the docu is mentioning spec files and Dockerfile for now.

File: xml/obs_build_job_constraints.xml [hunk 2]
```diff
@@ -19,6 +19,10 @@
  <para> The build constraints for a package are part of the package sources, as
   a _constraints XML source file (validated on submission). The _constraints
   source file can contain the values listed below. </para>
+ <para>In addition to that some build receipe parsers support to parse constraints
```
[the remark this answers is not in the document]
Comment (tomschr) [reply xml/obs_build_job_constraints.xml]: I think a noun is missing after "the above". Where do you refer to? To a list/paragraph/...? If it's the last para, you could use "previous paragraph".

If it's something that is not easy to describe or further away, use `<xref linkend="..."/>` to refer to the respective block element.

**Pull request merged**